ecpy_pulses/plugin.py

```diff
@@ -0,0 +1,855 @@
+                           for m in os.listdir(v_path)
+                           if (os.path.isfile(os.path.join(v_path, m)) and
+                               m.endswith('.enaml')))
+
+        if not os.path.isfile(os.path.join(pack_path, '__init__.py')):
+            log = logging.getLogger(__name__)
+            mess = cleandoc('''{} is not a valid Python package (miss
+                __init__.py).'''.format(pack + '.views'))
+            log.error(mess)
+            failed[pack] = mess
+            return [], []
+
+        for mod in v_modules[:]:
+            if mod in exceptions:
+                v_modules.remove(mod)
+
+        return modules, v_modules
+
+    def _explore_modules(self, modules, founds, mod_var, failed):
+        """ Explore a list of modules, looking for tasks.
+
+        Parameters
+        ----------
+        modules : list
+            The list of modules to explore
+
+        found : dict
+            A dict in which discovered objects will be stored.
+
+        mod_var : str
+            Name of the module variable to look for.
+
+        failed : list
+            A dict in which failed imports will be stored.
+
+        """
+        for mod in modules:
+            try:
+                m = import_module('.' + mod, MODULE_ANCHOR)
+            except Exception as e:
+                log = logging.getLogger(__name__)
+                mess = 'Failed to import {} : {}'.format(mod, e)
+                log.error(mess)
+                failed[mod] = mess
+                continue
+
+            if hasattr(m, mod_var):
+                var = getattr(m, mod_var)
+                if isinstance(var, list):
+                    founds.update({self._normalise_name(found.__name__): found
+                                  for found in var})
+                else:
+                    founds.update(var)
+
+    def _explore_views(self, modules, views, mod_var, failed):
+        """ Explore a list of modules, looking for views.
+
+        Parameters
+        ----------
+        modules : list
+            The list of modules to explore
+
+        views : dict
+            A dict in which discovered views will be stored.
+
+        mod_var : str
+            Name of the module variable to look for.
+
+        failed : list
+            A list in which failed imports will be stored.
+
+        """
+        for mod in modules:
+            try:
+                with enaml.imports():
+                    m = import_module('.' + mod, MODULE_ANCHOR)
+            except Exception as e:
+                log = logging.getLogger(__name__)
+                mess = 'Failed to import {} : {}'.format(mod, e)
+                log.error(mess)
+                failed[mod] = mess
+                continue
+
+            if hasattr(m, mod_var):
+                views.update(getattr(m, mod_var))
+
+    def _bind_observers(self):
+        """ Setup the observers for the plugin.
+
+        """
+        for folder in self.templates_folders:
```

Comment:
Should also update the sequences, shapes, etc.. list if the matching dictionaries are updated.